core/vm/instructions.go

```diff
@@ -465,23 +465,37 @@ func opGasprice(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([
 }
 
 func opBlockhash(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
```

Comment:
Better have 2 versions of this function: pre- and post-EIP-2935. You can do something similar to `opSelfdestruct6780` & `enable6780` (PR #7976).

Also, you need to something similar to `jt[SLOAD].dynamicGas = gasSLoadEIP2929` - see https://github.com/ethereum/EIPs/blob/72523ee3f865e09f8a6117c1b5e74cbb2df4f60e/EIPS/eip-2935.md#gas-costs-and-witnesses

Reply:
Done